drivers: pwm: mcux_ftm: allow to select clock source

manuargue opened the pull request:
Set RTC clock source to the internal 32 KHz LPO. Currently RTC clock is used to source RTC counter and FTM counter.
Fixes #71289

FTM internal counter can be clocked by one of three clock sources independent of the module bus clock. This patch introduces a DT property to perform the clock selection from DT.
DT sources are updated to keep the current clock selection for all boards, with exception of ucans32k1sic board which is migrated to use system by default, as this seems to be a better choice for most cases. Some PWM LED samples require slower clock so overlays are added for those cases.

manuargue-nxp committed:
dts: arm: nxp: s32k1xx: fix RTC clock source

Set RTC clock source to the internal 32 KHz LPO. Currently RTC clock is
used to source RTC counter and FTM counter.

Fixes #71289

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

decsny commented on drivers/pwm/pwm_mcux_ftm.c:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, NXP
+ * Copyright (c) 2017,2024 NXP
```
you can remove the (c) and put a space after the comma

decsny commented on dts/bindings/pwm/nxp,kinetis-ftm-pwm.yaml:
```diff
@@ -14,6 +14,18 @@ properties:
+      - "external"
+    description: |
+      Select one of three possible clock sources for the FTM counter. This clock
+      source selection is independent of the system clock source driving the FTM
```
should property name be more specific then?

manuargue-nxp committed:
drivers: pwm: mcux_ftm: allow to select clock source

FTM internal counter can be clocked by one of three clock sources
independent of the module bus clock. This patch introduces a DT property
to perform the clock selection from DT.

DT sources are updated to keep the current clock selection for all boards,
with exception of ucans32k1sic board which is migrated to use system
clock by default, as this seems to be a better choice for most cases.
Some PWM LED samples require slower clock so overlays are added for
those cases.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

Dat-NguyenDuy reviewed (dismissed):
LGTM 👍

mmahadevan108 commented:
Please update the SHA in `west.yml`

mmahadevan108 commented:
@anangl, could you kindly review.
